pkg/events/consumers/dnslookup.go

```diff
@@ -15,27 +16,76 @@ import (
 	"github.com/nats-io/nats.go"
 )
 
+// DNSLookupHandler struct contains attributes to handle worker pool pattern
 type DNSLookupHandler struct {
 	scanService interfaces.IScanService
+	workers     int            // number of workers
+	queue       chan *nats.Msg // jobQueue
+
+	// Metrics
+	processed  atomic.Uint64
+	failed     atomic.Uint64
+	dropped    atomic.Uint64
+	queueDepth atomic.Int32
 }
 
-func NewDNSLookupHandler(scanService interfaces.IScanService) *DNSLookupHandler {
-	return &DNSLookupHandler{scanService: scanService}
+// NewDNSLookupHandler contains also the worker number
+func NewDNSLookupHandler(scanService interfaces.IScanService, workers int) *DNSLookupHandler {
+	bufferSize := workers * 2
+	handler := &DNSLookupHandler{
+		scanService: scanService,
+		workers:     workers,
+		queue:       make(chan *nats.Msg, bufferSize), //  **BUFFERED** channel to avoid blocking
+	}
+	handler.startWorkers()
+	return handler
 }
 
 var _ interfaces.EventConsumer = (*DNSLookupHandler)(nil)
 
+// startWorkers launch all goroutines for workers, these are there until new work arrive
+func (h *DNSLookupHandler) startWorkers() {
+	for i := 0; i < h.workers; i++ {
+		go func() {
+			for msg := range h.queue {
+				// processing logic
+				// Update queue depth metric
+				h.queueDepth.Add(-1)
+				ctx, cancel := context.WithTimeout(context.Background(), 900*time.Second)
+				err := h.processDNSLookupEvent(ctx, msg.Data)
+				cancel()
+
+				if err != nil {
+					h.failed.Add(1)
+					slog.Debug("DNSLookupEvent incrementing metric of failed")
```

Comment:
`[Nit]` Spelling can be improved to make things clearer when debugging.
```suggestion
					slog.Debug("DNSLookupEventHandler: incrementing FAILED metric")
```